library/L1_Peripheral/lpc40xx/capture.hpp

```diff
@@ -0,0 +1,195 @@
+#pragma once
+#include <cstdint>
+#include "L0_Platform/lpc40xx/LPC40xx.h"
+#include "L1_Peripheral/interrupt.hpp"
+#include "L1_Peripheral/lpc40xx/pin.hpp"
+#include "L1_Peripheral/lpc40xx/system_controller.hpp"
+#include "L1_Peripheral/lpc40xx/timer.hpp"
+#include "L1_Peripheral/lpc40xx/capture.hpp"
+#include "L1_Peripheral/timer.hpp"
+#include "L1_Peripheral/capture.hpp"
+#include "utility/log.hpp"
+#include "utility/status.hpp"
+#include "utility/units.hpp"
+
+
+namespace sjsu
+{
+namespace lpc40xx
+{
+class Capture final : public sjsu::Capture
+{
+ public:
+    template<const Timer::ChannelPartial_t &port>
+    static void TimerHandler()
+    {
+        TimerHandler(port);
+    }
+
+    struct Channel // NOLINT
+    {
+     private:
+        inline static IsrPointer timer0_isr                 = nullptr;
+        inline static const Timer::ChannelPartial_t kTimerPartial0 = {
+        .timer_register = LPC_TIM0,
+        .power_id       = SystemController::Peripherals::kTimer0,
+        .irq            = IRQn::TIMER0_IRQn,
+        .user_callback  = &timer0_isr,
```

Comment:
If it is feasible, it would be nice to include a pin parameter here with the information needed to setup a pin for capture. We don't need to be able to control every pin, just a few. If the developer wants to use a pin we don't have they can construct their own Channel_t object.

Reply:
So restrict the configurations only to the two pins accessible on the SJ2 board? This ties into the other issue of multiple pins mapping to the same capture input depending on the lpc40xx package used (say LQFP80 vs LQFP100). Perhaps rather than having unique timer and channel parameters for the constructor there should be an enum that specifies the timer, channel, and pin -- that way the enum could only contain the valid combinations. But it seems a bit yucky too.